### PR TITLE
Use RP plugin events to decide warning to display

### PR DESCRIPTION
**Pull request by havocp**
Also, add a stub to watch these events during normal operation,
but for now this just logs them.

This doesn't quite work with the current RP plugin because it doesn't throw an error and stop the build from loading when typesafe.properties is missing. However, the new one that's about to come out will do that and then this should work.

I've also stubbed out here (just debug logged) the RP events for use after loading the project. However, I think this has a basic problem that we'll miss these events if we connect to sbt server after the build has already loaded. So, we may have to change this.

### CODE REVIEW

File: ui/app/assets/widgets/home/home.js [hunk 1]
```diff
@@ -18,18 +18,49 @@ define([
...
+        failedDueToNoTypesafeProperties = true;
+      else if (!event.serialized.subscriptionId)
+        failedDueToNoSubscriptionId = true;
+    }
+  });
+
+  subTypeEventStream("BuildFailedToLoad").each(function (e) {
+    var warning = null;
+
+    if (failedDueToNoTypesafeProperties) {
```
Comment (henrikengstrom) [inline ui/app/assets/widgets/home/home.js]: Will it always fail for one or the other (not both at the same time)?

File: ui/app/assets/widgets/home/home.js [hunk 2]
```diff
@@ -18,18 +18,49 @@ define([
...
-    });
+  var failedDueToNoTypesafeProperties = false;
+  var failedDueToNoSubscriptionId = false;
+
+  var sbtEventStream = websocket.subscribe('type','sbt');
+  var subTypeEventStream = function(subType) {
+    return sbtEventStream.matchOnAttribute('subType',subType);
+  };
+
+  subTypeEventStream("DetachedEvent").each(function(message) {
```
Comment (henrikengstrom) [inline ui/app/assets/widgets/home/home.js]: Can we always count that the ``DetachedEvent`` arrives before the ``BuildFailedToLoad`` event?

File: ui/app/assets/widgets/home/home.js (hunk 1 above)
Comment (havocp) [inline ui/app/assets/widgets/home/home.js]: yeah - in the SubscriptionIdEvent if fileExists=false then the subscriptionId is always None.

File: ui/app/assets/widgets/home/home.js (hunk 2 above)
Comment (havocp) [inline ui/app/assets/widgets/home/home.js]: The SubscriptionIdEvent is always before the plugin throws the exception that fails the load, so for that one yes. The SubscriptionLevelEvent and PlatformRelease event are sent async from a thread, so their order is not defined with respect to loading the build.

**Comment by henrikengstrom** [comment]
LGTM apart from comments/questions.

File: ui/app/assets/widgets/home/home.js (hunk 1 above)
Comment (henrikengstrom) [inline ui/app/assets/widgets/home/home.js]: If I'd just read the text below I could've figured out the flow...

**Pull request merged**